src/core/controllers/oidc.go

```diff
@@ -122,30 +123,100 @@ func (oc *OIDCController) Callback() {
 	}
 	oc.SetSession(tokenKey, tokenBytes)
 
+	oidcSettings, err := config.OIDCSetting()
+	if err != nil {
+		oc.SendInternalServerError(err)
+		return
+	}
+
 	if u == nil {
-		oc.SetSession(userInfoKey, string(ouDataStr))
-		oc.Controller.Redirect(fmt.Sprintf("/oidc-onboard?username=%s", strings.Replace(info.Username, " ", "_", -1)),
-			http.StatusFound)
-	} else {
-		gids, err := group.PopulateGroup(models.UserGroupsFromName(info.Groups, common.OIDCGroupType))
-		if err != nil {
-			log.Warningf("Failed to populate groups, error: %v, user will have empty group list, username: %s", err, info.Username)
-		}
-		u.GroupIDs = gids
-		oidcUser, err := dao.GetOIDCUserByUserID(u.UserID)
-		if err != nil {
-			oc.SendInternalServerError(err)
+		// Recover the username from d.Username by default
+		username := info.Username
```

Comment:
`username` may be empty string here?

Reply:
I'll test that use case again, but I think you get an error and onboard fails if the username is empty, which could both happen with a custom claim or with the default claim.

Reply:
After checking, there is not much difference from the previous code. info.Username could be empty string if the default username claim is empty.